galeracluster/source/scriptablesst.rst

```diff
@@ -60,70 +112,78 @@ Calling Conventions
 ----------------------------
 .. _`calling-conventions`:
 
-Scripts for state snapshot transfers should adhere to the following conventions.
+In writing your own custom script for state snapshot transfers, there are certain conventions that you need to follow in order to accommodate how Galera Cluster calls the script.
 
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Receiver
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-.. _`receiver`:
-
-On receiving side the script should accept the following positional arguments:
-
-- Role. This will be 'joiner'.
+.. _`call-receiver`:
 
-- Address to receive snapshot at. It will be either the value of `wsrep_sst_receive_address`` or some sensible default in the ``<ip>:<port>`` shape if the former is not set.
+When the node calls for a state snapshot transfer as a joiner, it begins by passing a number of arguments to the state transfer script, as defined in :ref:`Common Parameters <common-parameters>` above.  For your own script you can choose to use or ignore these arguments as suits your needs.
 
-- Authentication information as set in ``wsrep_sst_auth``.
+After the script receives these arguments, prepare the node to accept a state snapshot transfer.  For example, in the case of ``wsrep_sst_rsync.sh``, the script starts ``rsync`` in server mode.
 
-- The value of ``mysql_real_data_home`` (MySQL data directory path).
+To signal that the node is ready to receive the state transfer, print the following string to standard output: ``ready <address>:port\n``.  Use the IP address and port at which the node is waiting for the state snapshot.  For example:
 
-- Path to the configuration file, (``my.cnf`` or ``my.ini``, depending on your build).
+.. code-block:: console
 
-After script prepares the node for accepting state snapshot (e.g. ``wsrep_sst_rsync`` starts ``rsync`` in server mode), the script should print the following string to standard output::
+   ready 192.168.1.1:4444
 
-	ready <address>
+Next send a state transfer request to the donor node.  This is formed from the address and port of the joiner, the values given to ``--auth``, and the name of your script.  The donor node receives the request and uses these values as input parameters in running your script on that node.
```

Comment:
"Next send a state transfer" is not clear. It is a responsibility of the server to initiate that request and not the user or the script. See http://wiki.codership.com/doku.php?id=scriptable_state_snapshot_transfer

"After script prepares the node for accepting state snapshot (e.g. wsrep_sst_rsync starts rsync in server mode), the script should print the following string to standard output:

ready <address>
where <address> is the real address at which node is waiting for state transfer. This address, the value of wsrep_sst_auth, and the name of the script will be passed in a state transfer request to sender and will be part of input to sender script."